crates/air-lib/air-parser/src/ast/values.rs

```diff
@@ -54,13 +54,31 @@ pub struct StreamWithLambda<'i> {
     pub position: usize,
 }
 
+/// A canonicalized stream without lambda.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct CanonStream<'i> {
+    pub name: &'i str,
+    pub position: usize,
+}
+
+/// A canonicalized stream with lambda.
+#[derive(Debug, PartialEq, Eq, Clone, Serialize, Deserialize)]
+pub struct CanonStreamWithLambda<'i> {
+    pub name: &'i str,
```

Comment:
Generic question not quite related to this PR: `Deserialize` without `Cow` is very limited.  What is its propose on the AST types?